integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,164 @@
+        )
+        response.raise_for_status()
+        return response.json()
+
+    @cache_iterator_result()
+    async def get_clickup_teams(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all Clickup teams."""
+        url = f"{self.api_url}/team"
+        yield (await self._send_api_request(url)).get("teams", [])
+
+    @cache_iterator_result()
+    async def _get_spaces_in_team(
+        self, team_id: str
+    ) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all spaces in a workspace."""
+        url = f"{self.api_url}/team/{team_id}/space"
+        params = {"archived": "false"}
+        yield (await self._send_api_request(url, params)).get("spaces", [])
+
+    async def _get_folders_in_space(
+        self, team_id: str
+    ) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all folders in a space."""
+        async for spaces in self._get_spaces_in_team(team_id):
+            for space in spaces:
+                url = f"{self.api_url}/space/{space.get('id')}/folder"
+                params = {"archived": "false"}
+                yield (await self._send_api_request(url, params)).get("folders")
+
+    async def get_folder_projects(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all projects with a folder parent."""
+        async for teams in self.get_clickup_teams():
+            for team in teams:
+                async for folders in self._get_folders_in_space(team.get("id")):
+                    for folder in folders:
+                        url = f"{self.api_url}/folder/{folder.get('id')}/list"
+                        params = {"archived": "false"}
+                        response = await self._send_api_request(url, params)
+                        projects = response.get("lists")
+                        yield [
+                            {**project, TEAM_ID: team.get("id")} for project in projects
+                        ]
+
+    async def get_folderless_projects(
+        self,
+    ) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all projects without a folder parent."""
+        async for teams in self.get_clickup_teams():
+            for team in teams:
+                async for spaces in self._get_spaces_in_team(team.get("id")):
+                    for space in spaces:
+                        url = f"{self.api_url}/space/{space.get('id')}/list"
+                        params = {"archived": "false"}
+                        response = await self._send_api_request(url, params)
+                        projects = response.get("lists")
+                        yield [
+                            {**project, TEAM_ID: team.get("id")} for project in projects
+                        ]
+
+    async def get_single_project(self, list_id: str) -> dict[str, Any]:
+        """Get a single project by folder_id."""
+        url = f"{self.api_url}/list/{list_id}"
+        response = await self._send_api_request(url)
+        return response
```

Comment:
this will not have the team returned, meaning it will get override by an empty string once receive some update about the project

Reply:
I have included the team in the single response